### PR TITLE
[TM-1182] create a new version when edit polygon

### PR DESCRIPTION
**Pull request by LimberHope**
[Task Link](https://gfw.atlassian.net/browse/TM-1182)

### CODE REVIEW

File: src/components/elements/Map-mapbox/Map.tsx [hunk 1]
```diff
@@ -257,22 +262,56 @@ export const MapContainer = ({
...
       if (geojson) {
         if (polygonFromMap?.uuid) {
+          !pdView && onCancelEdit();
           const feature = geojson.features[0];
           const response = await fetchPutV2TerrafundPolygonUuid({
-            body: { geometry: JSON.stringify(feature) },
+            body: {
+              geometry: JSON.stringify(feature),
+              // @ts-ignore
+              adminUpdate: pdView ? false : true
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/Map.tsx:288]: pdView is already a boolean right? so no need to check

File: src/components/elements/Map-mapbox/Map.tsx [hunk 2]
```diff
@@ -257,22 +262,56 @@ export const MapContainer = ({
...
+            },
             pathParams: { uuid: polygonFromMap?.uuid }
           });
           reloadSiteData?.();
-          if (response.message == "Geometry updated successfully.") {
+          if (
+            response.message == "Site polygon version created successfully." ||
+            response.message == "Geometry updated successfully."
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/Map.tsx:295]: Please modify backend response, use status code 201 when an entity is created and 200 when an entity is updated.
Later come here and use those status codes for the conditional.
Right now it is depending in synchronizing the messages between BE and FE

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/Map-mapbox/components/OverviewMapArea.tsx [hunk 3]
```diff
@@ -205,6 +205,7 @@ const OverviewMapArea = ({
         polygonFromMap={polygonFromMap}
         shouldBboxZoom={!shouldRefetchPolygonData}
         modelFilesData={modelFilesData?.data}
+        pdView={true}
```
Comment (cesarLima1) [inline src/components/elements/Map-mapbox/components/OverviewMapArea.tsx:208]: Just some question:
Did you test this with users who are not Monitoring Partners? What should happen with them? Are they able to create versions?

Comment (LimberHope) [reply src/components/elements/Map-mapbox/components/OverviewMapArea.tsx:208]: If you are not a project monitoring partner, you should not have access to the project or those functions.

**Review by cesarLima1 — DISMISSED** [review]

File: src/components/elements/Map-mapbox/Map.tsx (hunk 2 above)
Comment (roguenet) [reply src/components/elements/Map-mapbox/Map.tsx]: Why do we have an endpoint that can both create and update? These are separate REST actions and should be separate endpoints.

**Review by roguenet — APPROVED** [review]

File: src/components/elements/Map-mapbox/Map.tsx [hunk 4]
```diff
@@ -257,23 +263,59 @@ export const MapContainer = ({
...
+      return;
+    }
+    map.current.fitBounds(bounds, {
+      padding: 100,
+      linear: false
+    });
+  };
+
+  const { mutate: putMutate } = usePutV2TerrafundPolygonUuid();
+  const { mutate: postMutate } = usePostV2GeometryUUIDNewVersion();
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/Map.tsx:279]: rename this two, I suggest `updateGeometry` and `createGeometry`, but up to you, I just find very hard to understand the put and post mutate

**Review by pachonjcl — CHANGES_REQUESTED** [review]

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**